include/psa/crypto.h

```diff
@@ -3314,12 +3358,31 @@ psa_status_t psa_key_derivation_input_bytes(
  * \param step                    Which step the input data is for.
  * \param key                     Identifier of the key. It must have an
  *                                appropriate type for step and must allow the
- *                                usage #PSA_KEY_USAGE_DERIVE.
+ *                                usage #PSA_KEY_USAGE_DERIVE or
```

Comment:
I wonder if the specification was overly restrictive here. The _secret_ (`PSA_KEY_DERIVATION_INPUT_SECRET`) input to a key derivation should be a suitable key, i.e. it should have the type `PSA_KEY_TYPE_DERIVE` for `input_key` or a key agreement type for `key_agreement`, and should have the usage `PSA_KEY_USAGE_DERIVE`. But it can be useful to take a key object as a salt/label/… input, too, regardless of whether it allows `USAGE_DERIVE`.

However not all keys should be usable that way. For example, passwords must not be usable in this way, since it would amount to allowing them to be hashed by a non-stretching key derivation!

Reply:
I think the key type avoids that. Currently `input_key()` requires the type of the key to be compatible with the step, and the only step that accepts a key of type `PASSWORD` is `INPUT_PASSWORD`. Other steps accept keys or type `RAW_DATA`, or `PEPPER` for the salt step.

Reply:
Right. But a salt still has to have `USAGE_DERIVE`, which I'm not convinced is a useful requirement.

Reply:
I agree, I'm not convinced it's useful either, but I don't think it's hurtful either (that's what my previous comment was about), I didn't think about it enough to be fully 100% convinced it's not useful, it's pre-existing, and lifting requirements is a backwards-compatible change that can be done later, so I'd rather keep it as is in this PR and leave it for later. Would you object to that?